### PR TITLE
[EA Forum only] use event cards on the group details page

### PR DESCRIPTION
**Pull request by s-cheng**
Since we created event cards for the Events page, we are now updating the group details page to list the group's events using those cards as well.

Screen Shot 2022-03-02 at 7 22 06 PM

Past events are desaturated, and events without a start date (very rare) get their own section, due to the fact that mongo won't let us order them at the end of the "Upcoming Events" list.

Screen Shot 2022-03-02 at 7 23 17 PM

### CODE REVIEW

File: packages/lesswrong/components/localGroups/LocalGroupPage.tsx [hunk 1]
```diff
@@ -139,6 +209,73 @@ const LocalGroupPage = ({ classes, documentId: groupId }: {
       mapOptions={{zoom:11, center: group.googleLocation.geometry.location, initialOpenWindows:[groupId]}}
     />
   }
+  
+  // the EA Forum shows the group's events as event cards instead of post list items
+  let upcomingEventsList = <PostsList2 terms={{view: 'upcomingEvents', groupId: groupId}} />
+  if (isEAForum) {
```
Comment (jpaddison3) [inline packages/lesswrong/components/localGroups/LocalGroupPage.tsx:215]: I would consider making a feature flag for "hasNewEvents"

File: packages/lesswrong/components/posts/PostsList2.tsx [hunk 2]
```diff
@@ -154,12 +154,13 @@ const PostsList2 = ({
   useOnMountTracking({eventType: "postList", eventProps: {postIds, hidePosts}, captureOnMount: eventProps => eventProps.postIds.length, skip: !postIds.length||loading})
 
   if (!orderedResults && loading) return <Loading />
+  if (results && !results.length && !showNoResults) return null
```
Comment (jpaddison3) [inline packages/lesswrong/components/posts/PostsList2.tsx:157]: Have you tested at least one other location in the site where showNoResults is used?

**Comment by jpaddison3** [comment]
I'm assuming you verified that the LW version looks the same.

**Review by jpaddison3 — APPROVED** [review]
ASTLAC

File: packages/lesswrong/components/posts/PostsList2.tsx (hunk 2 above)
Comment (s-cheng) [reply packages/lesswrong/components/posts/PostsList2.tsx:157]: Yeah, it's only used in a couple places - I tested and they looked unaffected to me

**Pull request merged**